src/prompt/functions/stringify-for-model.ts

```diff
@@ -0,0 +1,14 @@
+import type { Jsonifiable } from 'type-fest';
+
+/**
+ * Stringifies a JSON value in a way that's optimized for use with LLM prompts.
+ *
+ * This is intended to be used with `function` and `tool` arguments and responses.
+ */
+export function stringifyForModel(jsonObject: Jsonifiable | void): string {
+  if (jsonObject === undefined) {
+    return '';
+  }
+
+  return JSON.stringify(jsonObject, null, 0);
```

Comment:
i commented elsewhere, but i think this too restrictive given our confidence in the impact.